_posts/en/newsletters/2018-11-13-newsletter.md

```diff
@@ -0,0 +1,141 @@
+mailing list and suggests an opportunity to develop a new tool some
+users would find helpful.  Several notable code changes in popular
+Bitcoin infrastructure projects are also described.
+
+## Action items
+
+None this week.
+
+## News
+
+- **LN developer summit and mailing list activity:** before, during, and
+  after a planned meeting among Lightning Network protocol developers,
+  the [Lightning-Dev mailing list][] saw a surge of new proposals and
+  discussion about earlier proposals.  Below are some highlights:
+
+    - **Advertising node liquidity:** Lisa Neigut [proposes][neigut
+      liquidity] allowing LN nodes advertise that they're willing to
+      provide incoming capacity in exchange for a certain level of fees.
+      Merchants need their payment channels to have incoming capacity in
+      order to be able receive secure offchain payments from
+      customers---the current alternatives are either requiring some of
+      their customers to wait for several onchain confirmations to open
+      a new channel or making manual channel liquidity arrangements with
+      other merchants.  Although solving this problem would be highly
+      advantageous for merchant adoption of LN, it does pose some
+      technical challenges that discussion participants attempt to solve
+      both in this thread and in a [related thread][zmn liquidity].
+
+    - **Walletless channel opens:** pseudonymous contributor ZmnSCPxj
+      [notes][walletless opens] that if [BIP118][]
+      SIGHASH_NOINPUT_UNSAFE were adopted to allow implementing the
+      [Eltoo protocol][] in LN, then an LN implementation could set up a
+      channel with a peer offchain and simply provide a regular Bitcoin
+      wallet with an address to pay in order to open the channel.  The
+      LN implementation could then handle all of the secure offchain
+      payments and ensure that the channel closing balance is sent to
+      one of the regular wallet's addresses.
+
+    - **Making path probing more convenient:** Anthony Towns
+      [proposes][probe cancel] a method for allowing all the nodes along
+      a path to forget about a small-value payment if one of the nodes
+      on the path is offline.  This reduces the resources required in
+      the case of a routing failure by a node that proactively probes
+      its available payment paths to determine which are the fastest and
+      most reliable for sending payments.
+
+- **Opportunity available for providing utility functions outside of
+  Bitcoin Core:** Bitcoin Core's RPC interface currently provides over
+  100 methods and there are often proposals to add even more for utility
+  functions that don't require access to the internal state of the node
+  or wallet.  During last week's developer IRC [meeting][core dev
+  meeting], members of the project reaffirmed their commitment to not
+  providing any new utility functions under normal circumstances in
```

Comment:
I think this part "not providing any new utility functions" and below "the ability to gracefully deprecate those options" are putting it a bit strongly. See sipa and wumpus from the meeting:

19:16:18 <sipa> i think that bitcoin core, for better or worse, implements certain featuresets - and it makes sense to implement those features completely, even if some are stateless
...
19:17:19 <sipa> [...] signrawtransaction makes perfect sense, as we offer raw transaction functionality

19:15:12 <wumpus> I don't think the current ones should be removed, but i'm not sure about adding new ones
19:24:45 <wumpus> I'm not arguing to remove the current ones or so, just prevent an onslaught of new ones

sipa is saying for certain utilities it makes sense to offer them as an RPC even if they are stateless, and wumpus is saying that the goal isn't to deprecate existing RPCs, just to prevent feature creep.

Reply:
I'd recommend just removing the "giving the Bitcoin Core project the ability to gracefully deprecate those options and recommend the new tool to users needing those functions." part.

Reply:
To be fair, I wrote "not providing any new utility functions **under normal circumstances**" :-)  sipa gave specific reasons in the meeting for when he thought they'd be ok to add (and provided a specific example).  I also think there were comments in the meeting about wanting to be able to deprecate at least some of the utility functions, e.g.

> <sipa> i don't really care for decodescript being in our codebase really - it's trivial to do anywhere

That said, I think you're right about removing that final clause.  I'll do that.

Reply:
I've edited this both per your suggestion and a bit extra too so I think the current text should be a quite conservative reading of the meeting discussion.